indexd/index/drivers/alchemy.py

```diff
@@ -1029,6 +1029,31 @@ def get_latest_version(self, did, has_version=None):
 
             return record.to_document_dict()
 
+    def bulk_get_latest_versions(self, dids, skip_null=False):
+        """
+        Get latest version given list of did
+        """
+        with self.session as session:
+            # get the baseid from given dids to filter the maxdate subquery
+            subq = session.query(IndexRecord.baseid).filter(IndexRecord.did.in_(dids))
+
+            # get max date for each baseid
+            max_date_subq = session.query(IndexRecord.baseid, func.max(IndexRecord.created_date).label('max_date')) \
+                .filter(IndexRecord.baseid.in_(subq)) \
+                .group_by(IndexRecord.baseid)
+
+            if skip_null:
+                max_date_subq = max_date_subq.filter(IndexRecord.version.isnot(None))
+
+            max_date_subq = max_date_subq.subquery()
+
+            # get the index record with thhe latest create date for each baseid
+            query = session.query(IndexRecord).join(max_date_subq, and_(
+                max_date_subq.c.baseid == IndexRecord.baseid,
+                max_date_subq.c.max_date == IndexRecord.created_date
+            ))
+            return [q.to_document_dict() for q in query]
```

Comment:
How are we relating the new versions to the old UUIDs?

Reply:
I decided to leave that for the client to do by filtering by baseid. So for gdcapi download it's calling indexd twice now, and map the two list of docs by baseid.

I had an indexd endpoint that would add doc.latest_id in the response. I removed it and use this simpler approach since in gdcapi download it's possible a user would request downloading the latest released version of any uuid, but the doc with the uuid that exist in the graph is the latest any version (including null). So to achieve single call we would tweak indexd further and it kind of becomes too specific.